lib/src/util/dart_type_utilities.dart

```diff
@@ -47,13 +47,33 @@ class DartTypeUtilities {
   static bool hasInheritedMethod(MethodDeclaration node) =>
       lookUpInheritedMethod(node) != null;
 
+  static InterfaceType implementedInterface(
+      DartType type, String interface, String library) {
+    if (type is! InterfaceType) {
+      return null;
+    }
+    bool predicate(InterfaceType i) => isInterface(i, interface, library);
+
+    ClassElement element = type.element;
+    if (predicate(type)) {
+      return type;
+    }
+    if (!element.isSynthetic && type is InterfaceType) {
+      final interfaces = element.allSupertypes.where(predicate);
```

Comment:
Fixed

@bwilkerson I resolved it doing a DFS with the supertypes and propagating the typeArguments with their respective typeParameter to the super classes and implemented interfaces then I took the value of the class Iterable with their propagated types to compare and lint the node, PTAL.